### PR TITLE
[#1576] Android 15 memory page size 16kb support

### PR DESCRIPTION
**Pull request by Milan-Cerovsky**
# Author

- [x] **Self-review** your own code in GitHub's web interface[^1]
- [ ] Add **automated tests** as appropriate
- [ ] Update the [**manual tests**](../blob/main/docs/testing/manual_testing)[^2] as appropriate
- [ ] Check the **code coverage**[^3] report for the automated tests
- [x] Update **documentation** as appropriate (e.g [README.md](../blob/main/README.md), [Architecture.md](../blob/main/docs/Architecture.md), etc.)
- [x] **Run the demo app** and try the changes
- [x] Pull in the latest changes from the **main** branch and **squash** your commits before assigning a reviewer[^4]

# Reviewer

- [x] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
- [x] Perform an **ad hoc review**[^5]
- [ ] Review the **automated tests**
- [ ] Review the **manual tests**
- [x] Review the **documentation**, [README.md](../blob/main/README.md), [Architecture.md](/blob/main/docs/Architecture.md), etc. as appropriate
- [x] **Run the demo app** and try the changes[^6]

[^1]: _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
[^2]: _While we aim for automated testing of the SDK, some aspects require manual testing. If you had to manually test 
something during development of this pull request, write those steps down._
[^3]: _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed. Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules._
[^4]: _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._
[^5]: _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
[^6]: _While the CI server runs the demo app to look for build failures or crashes, humans running the demo app are 
more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

File: backend-lib/build.gradle.kts [hunk 1]
```diff
@@ -84,6 +84,9 @@ cargo {
     )
     profile = "release"
     prebuiltToolchains = true
+    exec = { spec, _ ->
```
Comment (Milan-Cerovsky) [inline backend-lib/build.gradle.kts:87]: Tells the compiler to use given page size

please refer to https://github.com/mozilla/rust-android-gradle/blob/c62655cae09dc7b7540e96a07d35881f18ce4251/plugin/src/main/kotlin/com/nishtahir/CargoBuildTask.kt#L231C34-L231C65

**Review by HonzaR — COMMENTED** [review]
I just needed to bump my dev environment NDK version to the same version.

**Review by HonzaR — APPROVED** [review]

File: backend-lib/build.gradle.kts [hunk 2]
```diff
@@ -84,6 +84,9 @@ cargo {
...
+        spec.environment["RUST_ANDROID_GRADLE_CC_LINK_ARG"] = "-Wl,-z,max-page-size=16384"
```
Comment (str4d) [inline backend-lib/build.gradle.kts:88]: I would expect this to break compilation, as it is overwriting the environment variable from the cargo build task to not include `-soname`. Does it not do so?

Comment (HonzaR) [reply backend-lib/build.gradle.kts:88]: It's working for me after this change. Tested on emulators Android SDK API levels 35 and 27 by creating a new wallet, syncing, and retrieving addresses or balances.

Comment (str4d) [reply backend-lib/build.gradle.kts:88]: Well if it works then great.

**Review by HonzaR — APPROVED** [review]

**Pull request merged**